tests/manage/monitoring/conftest.py

```diff
@@ -273,48 +300,24 @@ def corrupt_pg():
         minutest to start firing.
 
         Returns:
-            str: Name of corrupted deployment
+            str: Name of corrupted pod
         """
         # run_time of operation
-        run_time = 60 * 12
-        nonlocal oc
+        run_time = 60 * 14
         nonlocal pool_name
         nonlocal pool_object
-        nonlocal dummy_pod
-        nonlocal pg
         nonlocal osd_deployment
-        nonlocal dummy_deployment
 
-        logger.info(f"Corrupting {pg} PG on {osd_deployment}")
-        dummy_pod.exec_sh_cmd_on_pod(
-            f"ceph-objectstore-tool --data-path /var/lib/ceph/osd/ceph-"
-            f"{osd_deployment.split('-')[-1]} --pgid {pg} {pool_object} "
-            f"set-bytes /etc/shadow --no-mon-config"
-        )
-        logger.info("Unsetting osd noout flag")
-        ct_pod.exec_ceph_cmd("ceph osd unset noout")
-        ct_pod.exec_ceph_cmd(f"ceph pg deep-scrub {pg}")
-        oc.exec_oc_cmd(f"scale --replicas=0 deployment/{dummy_deployment}")
-        oc.exec_oc_cmd(f"scale --replicas=1 deployment/{osd_deployment}")
+        logger.info(f"Corrupting pool {pool_name} on {osd_deployment.name}")
+        rados_utils.corrupt_pg(osd_deployment, pool_name, pool_object)
         logger.info(f"Waiting for {run_time} seconds")
         time.sleep(run_time)
-        return osd_deployment
+        return osd_deployment.name
 
     test_file = os.path.join(measurement_dir, "measure_corrupt_pg.json")
     measured_op = measure_operation(corrupt_pg, test_file)
-    logger.info(f"Deleting pool {pool_name}")
-    ct_pod.exec_ceph_cmd(
-        f"ceph osd pool delete {pool_name} {pool_name} "
-        f"--yes-i-really-really-mean-it"
-    )
-    logger.info(f"Checking that pool {pool_name} is deleted")
 
-    logger.info(f"Deleting deployment {dummy_deployment}")
-    oc.delete(resource_name=dummy_deployment)
-
-    # wait for ceph to return into HEALTH_OK state after osd deployment
-    # is returned back to normal
-    ceph_health_check(tries=20, delay=15)
+    teardown()
```

Comment:
What is the purpose of calling the `teardown` here?

Reply:
Local teardown function returns cluster into a healthy state it is called here to mitigate all errors after the measurement.

Reply:
Okay, I see.